Source/WebCore/html/HTMLTextFormControlElement.cpp

```diff
@@ -907,6 +907,9 @@ void HTMLTextFormControlElement::adjustInnerTextStyle(const RenderStyle& parentS
         }
     }
 
+    if (parentStyle.fieldSizing() == FieldSizing::Content)
+        textBlockStyle.setLogicalMinWidth(Length { caretWidth(), LengthType::Fixed });
```

Comment:
Is there a way for this to be tested in Web Platform Tests instead of just in WebKit-specific tests? It seems like the kind of thing we want to ensure is consistent across browsers.

Reply:
I'm not familiar with anyway to get the size of a caret using exposed web platform APIs.

Reply:
Right, the caret width differs (at the least) on Cocoa platforms, so any test would need to have a way to directly/indirectly get the width.

Interestingly, this patch might be the first place to (indirectly) expose caret width to the web?

Could we still write a WPT that asserts a greater-than-or-equal-to width, in addition to keeping this WebKit-specific test?

Reply:
It looks like there is specified API to get information about the caret, `CaretPosition.getClientRect()`, but it's only implemented in Firefox.

https://drafts.csswg.org/cssom-view/#dom-caretposition-getclientrect

Reply:
One other reason why we might not want to add a WPT for this, is that the spec only says that agents "may" take the caret width into account. It doesn't require it? https://html.spec.whatwg.org/#the-textarea-element-2

Reply:
Open ended specifications like that are not great for interoperability!

Reflecting on it, I think space for the caret is needed alongside the text, so the **minimum** size isn’t all that’s needed, if someone is trying to fit a textarea to the size of the text that will be inside it.

Reply:
> Open ended specifications like that are not great for interoperability!

I agree fwiw and I don't think there's a reason a UA shouldn't account for the caret width as a minimum.

> Reflecting on it, I think space for the caret is needed alongside the text

The only situation where there appears to be an issue with hidden carets is when the content size is 0px (an empty input or textarea). When there's text in the control the caret shows fine from my testing. I'd also personally prefer to match Chromium as close as possible and then raise any spec issues necessary to reach improvements if they're needed.

Reply:
Am I okay to get this PR merged or would you prefer me to add a minimum caret width test to WPT?

Reply:
Oh that was never intended to block merging this change in. If the concerns about the web platform and fingerprinting are not serious then I don't know of any blocker.